Custom providers #104 (#105)
ADmad/cakephp-social-auth: README.md

EventManager::instance()->on(new SocialAuthListener());
```

### Extend with custom providers

> In order to enable custom or not preconfigured provideres you can extend the middleware configuration with *collectionFactory* and create your own instance of *SocialConnect\Auth\CollectionFactory*.

Reviewer: ```suggestion
In order to enable custom or not preconfigured providers you can extend the middleware configuration with `collectionFactory` and create your own instance of `SocialConnect\Auth\CollectionFactory`.
```

Attach the listener in your `Application` class:

```php
// src/Application.php
SocialConnect\Auth\CollectionFactory;

// Instantiate your own CollectionFactory with
...
public function middleware($middlewareQueue): \Cake\Http\MiddlewareQueue
...
$cf = new CollectionFactory();
$cf->register(\App\Authenticator\CustomProvider::NAME, \App\Authenticator\CustomProvider::class);
...

// Add this to your configuration
'collectionFactory' => $cf,

// src/Authenticator\CustomProvider.php
/*

>     You will find a lot of examples invendor/socialconnect/auth/src/OpenIDConnect/Provider/

Reviewer: ```suggestion
    You will find a lot of examples in vendor/socialconnect/auth/src/OpenIDConnect/Provider/
```

*/
```

> 

Reviewer: ```suggestion
```

EventManager::instance()->on(new SocialAuthListener());
```

### Extend with custom providers

In order to enable custom or not preconfigured provideres you can extend the middleware configuration with *serviceConfig.collectionFactory* and create your own instance of *SocialConnect\Auth\CollectionFactory*.

Attach the listener in your `Application` class:

```php
// src/Application.php
SocialConnect\Auth\CollectionFactory;

// Instantiate your own CollectionFactory with
...
public function middleware($middlewareQueue): \Cake\Http\MiddlewareQueue
...
$cf = new CollectionFactory();
$cf->register(\App\Authenticator\CustomProvider::NAME, \App\Authenticator\CustomProvider::class);
...

// Add this to your configuration

> serviceConfig.collectionFactory => $cf,

Reviewer: Don't nest `collectionFactory` under `serviceConfig`. Keep it as a separate top level config. Additional key inside the service options argument could cause problems in future.

Author: yes, moved
